### ISSUE TITLE
Issue on windows 10 Python 3.6

### ISSUE THREAD

**Issue by MatthieuDartiailh** [tartley/colorama#119]
Running pytest 3.0.5 and colorama 0.3.7 on Python 3.6 on a windows 10 machine I got the following error. I think that Python 3.6 made some change to console handling on windows so it may be related.
```
Microsoft Windows [version 10.0.14393]

Traceback (most recent call last):
  File "D:\Coding\Languages\Python\envs\py36\lib\site-packages\_pytest\main.py", line 98, in wrap_session
    session.exitstatus = doit(config, session) or 0
  File "D:\Coding\Languages\Python\envs\py36\lib\site-packages\_pytest\main.py", line 133, in _main
    config.hook.pytest_runtestloop(session=session)
  File "D:\Coding\Languages\Python\envs\py36\lib\site-packages\_pytest\vendored_packages\pluggy.py", line 745, in __call__
    return self._hookexec(self, self._nonwrappers + self._wrappers, kwargs)
  File "D:\Coding\Languages\Python\envs\py36\lib\site-packages\_pytest\vendored_packages\pluggy.py", line 339, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
  File "D:\Coding\Languages\Python\envs\py36\lib\site-packages\_pytest\vendored_packages\pluggy.py", line 334, in <lambda>
    _MultiCall(methods, kwargs, hook.spec_opts).execute()
  File "D:\Coding\Languages\Python\envs\py36\lib\site-packages\_pytest\vendored_packages\pluggy.py", line 614, in execute
    res = hook_impl.function(*args)
  File "D:\Coding\Languages\Python\envs\py36\lib\site-packages\_pytest\main.py", line 154, in pytest_runtestloop
    item.config.hook.pytest_runtest_protocol(item=item, nextitem=nextitem)
  File "D:\Coding\Languages\Python\envs\py36\lib\site-packages\_pytest\vendored_packages\pluggy.py", line 745, in __call__
    return self._hookexec(self, self._nonwrappers + self._wrappers, kwargs)
  File "D:\Coding\Languages\Python\envs\py36\lib\site-packages\_pytest\vendored_packages\pluggy.py", line 339, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
  File "D:\Coding\Languages\Python\envs\py36\lib\site-packages\_pytest\vendored_packages\pluggy.py", line 334, in <lambda>
    _MultiCall(methods, kwargs, hook.spec_opts).execute()
  File "D:\Coding\Languages\Python\envs\py36\lib\site-packages\_pytest\vendored_packages\pluggy.py", line 613, in execute
    return _wrapped_call(hook_impl.function(*args), self.execute)
  File "D:\Coding\Languages\Python\envs\py36\lib\site-packages\_pytest\vendored_packages\pluggy.py", line 254, in _wrapped_call
    return call_outcome.get_result()
  File "D:\Coding\Languages\Python\envs\py36\lib\site-packages\_pytest\vendored_packages\pluggy.py", line 279, in get_result
    raise ex[1].with_traceback(ex[2])
  File "D:\Coding\Languages\Python\envs\py36\lib\site-packages\_pytest\vendored_packages\pluggy.py", line 265, in __init__
    self.result = func()
  File "D:\Coding\Languages\Python\envs\py36\lib\site-packages\_pytest\vendored_packages\pluggy.py", line 614, in execute
    res = hook_impl.function(*args)
  File "D:\Coding\Languages\Python\envs\py36\lib\site-packages\_pytest\runner.py", line 66, in pytest_runtest_protocol
    runtestprotocol(item, nextitem=nextitem)
  File "D:\Coding\Languages\Python\envs\py36\lib\site-packages\_pytest\runner.py", line 79, in runtestprotocol
    reports.append(call_and_report(item, "call", log))
  File "D:\Coding\Languages\Python\envs\py36\lib\site-packages\_pytest\runner.py", line 137, in call_and_report
    hook.pytest_runtest_logreport(report=report)
  File "D:\Coding\Languages\Python\envs\py36\lib\site-packages\_pytest\vendored_packages\pluggy.py", line 745, in __call__
    return self._hookexec(self, self._nonwrappers + self._wrappers, kwargs)
  File "D:\Coding\Languages\Python\envs\py36\lib\site-packages\_pytest\vendored_packages\pluggy.py", line 339, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
  File "D:\Coding\Languages\Python\envs\py36\lib\site-packages\_pytest\vendored_packages\pluggy.py", line 334, in <lambda>
    _MultiCall(methods, kwargs, hook.spec_opts).execute()
  File "D:\Coding\Languages\Python\envs\py36\lib\site-packages\_pytest\vendored_packages\pluggy.py", line 614, in execute
    res = hook_impl.function(*args)
  File "D:\Coding\Languages\Python\envs\py36\lib\site-packages\_pytest\terminal.py", line 207, in pytest_runtest_logreport
    self.write_fspath_result(rep.nodeid, letter)
  File "D:\Coding\Languages\Python\envs\py36\lib\site-packages\_pytest\terminal.py", line 123, in write_fspath_result
    self._tw.write(res)
  File "D:\Coding\Languages\Python\envs\py36\lib\site-packages\py\_io\terminalwriter.py", line 198, in write
    write_out(self._file, markupmsg)
  File "D:\Coding\Languages\Python\envs\py36\lib\site-packages\py\_io\terminalwriter.py", line 333, in write_out
    fil.write(msg)
  File "D:\Coding\Languages\Python\envs\py36\lib\site-packages\colorama\ansitowin32.py", line 40, in write
    self.__convertor.write(text)
  File "D:\Coding\Languages\Python\envs\py36\lib\site-packages\colorama\ansitowin32.py", line 141, in write
    self.write_and_convert(text)
  File "D:\Coding\Languages\Python\envs\py36\lib\site-packages\colorama\ansitowin32.py", line 169, in write_and_convert
    self.write_plain_text(text, cursor, len(text))
  File "D:\Coding\Languages\Python\envs\py36\lib\site-packages\colorama\ansitowin32.py", line 175, in write_plain_text
    self.wrapped.flush()
OSError: [WinError 87] Paramètre incorrect

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "D:\Coding\Languages\Python\envs\py36\lib\site-packages\_pytest\main.py", line 111, in wrap_session
    config.notify_exception(excinfo, config.option)
  File "D:\Coding\Languages\Python\envs\py36\lib\site-packages\_pytest\config.py", line 923, in notify_exception
    excinfo=excinfo)
  File "D:\Coding\Languages\Python\envs\py36\lib\site-packages\_pytest\vendored_packages\pluggy.py", line 745, in __call__
    return self._hookexec(self, self._nonwrappers + self._wrappers, kwargs)
  File "D:\Coding\Languages\Python\envs\py36\lib\site-packages\_pytest\vendored_packages\pluggy.py", line 339, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
  File "D:\Coding\Languages\Python\envs\py36\lib\site-packages\_pytest\vendored_packages\pluggy.py", line 334, in <lambda>
    _MultiCall(methods, kwargs, hook.spec_opts).execute()
  File "D:\Coding\Languages\Python\envs\py36\lib\site-packages\_pytest\vendored_packages\pluggy.py", line 614, in execute
    res = hook_impl.function(*args)
  File "D:\Coding\Languages\Python\envs\py36\lib\site-packages\_pytest\terminal.py", line 164, in pytest_internalerror
    self.write_line("INTERNALERROR> " + line)
  File "D:\Coding\Languages\Python\envs\py36\lib\site-packages\_pytest\terminal.py", line 145, in write_line
    self.ensure_newline()
  File "D:\Coding\Languages\Python\envs\py36\lib\site-packages\_pytest\terminal.py", line 136, in ensure_newline
    self._tw.line()
  File "D:\Coding\Languages\Python\envs\py36\lib\site-packages\py\_io\terminalwriter.py", line 203, in line
    self.write('\n')
  File "D:\Coding\Languages\Python\envs\py36\lib\site-packages\py\_io\terminalwriter.py", line 198, in write
    write_out(self._file, markupmsg)
  File "D:\Coding\Languages\Python\envs\py36\lib\site-packages\py\_io\terminalwriter.py", line 333, in write_out
    fil.write(msg)
  File "D:\Coding\Languages\Python\envs\py36\lib\site-packages\colorama\ansitowin32.py", line 40, in write
    self.__convertor.write(text)
  File "D:\Coding\Languages\Python\envs\py36\lib\site-packages\colorama\ansitowin32.py", line 141, in write
    self.write_and_convert(text)
  File "D:\Coding\Languages\Python\envs\py36\lib\site-packages\colorama\ansitowin32.py", line 169, in write_and_convert
    self.write_plain_text(text, cursor, len(text))
  File "D:\Coding\Languages\Python\envs\py36\lib\site-packages\colorama\ansitowin32.py", line 174, in write_plain_text
    self.wrapped.write(text[start:end])
OSError: [WinError 87] Paramètre incorrect

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "D:\Coding\Languages\Python\envs\py36\Scripts\pytest-script.py", line 5, in <module>
    sys.exit(pytest.main())
  File "D:\Coding\Languages\Python\envs\py36\lib\site-packages\_pytest\config.py", line 57, in main
    return config.hook.pytest_cmdline_main(config=config)
  File "D:\Coding\Languages\Python\envs\py36\lib\site-packages\_pytest\vendored_packages\pluggy.py", line 745, in __call__
    return self._hookexec(self, self._nonwrappers + self._wrappers, kwargs)
  File "D:\Coding\Languages\Python\envs\py36\lib\site-packages\_pytest\vendored_packages\pluggy.py", line 339, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
  File "D:\Coding\Languages\Python\envs\py36\lib\site-packages\_pytest\vendored_packages\pluggy.py", line 334, in <lambda>
    _MultiCall(methods, kwargs, hook.spec_opts).execute()
  File "D:\Coding\Languages\Python\envs\py36\lib\site-packages\_pytest\vendored_packages\pluggy.py", line 614, in execute
    res = hook_impl.function(*args)
  File "D:\Coding\Languages\Python\envs\py36\lib\site-packages\_pytest\main.py", line 127, in pytest_cmdline_main
    return wrap_session(config, _main)
  File "D:\Coding\Languages\Python\envs\py36\lib\site-packages\_pytest\main.py", line 122, in wrap_session
    exitstatus=session.exitstatus)
  File "D:\Coding\Languages\Python\envs\py36\lib\site-packages\_pytest\vendored_packages\pluggy.py", line 745, in __call__
    return self._hookexec(self, self._nonwrappers + self._wrappers, kwargs)
  File "D:\Coding\Languages\Python\envs\py36\lib\site-packages\_pytest\vendored_packages\pluggy.py", line 339, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
  File "D:\Coding\Languages\Python\envs\py36\lib\site-packages\_pytest\vendored_packages\pluggy.py", line 334, in <lambda>
    _MultiCall(methods, kwargs, hook.spec_opts).execute()
  File "D:\Coding\Languages\Python\envs\py36\lib\site-packages\_pytest\vendored_packages\pluggy.py", line 613, in execute
    return _wrapped_call(hook_impl.function(*args), self.execute)
  File "D:\Coding\Languages\Python\envs\py36\lib\site-packages\_pytest\vendored_packages\pluggy.py", line 250, in _wrapped_call
    wrap_controller.send(call_outcome)
  File "D:\Coding\Languages\Python\envs\py36\lib\site-packages\_pytest\terminal.py", line 359, in pytest_sessionfinish
    self._tw.line("")
  File "D:\Coding\Languages\Python\envs\py36\lib\site-packages\py\_io\terminalwriter.py", line 203, in line
    self.write('\n')
  File "D:\Coding\Languages\Python\envs\py36\lib\site-packages\py\_io\terminalwriter.py", line 198, in write
    write_out(self._file, markupmsg)
  File "D:\Coding\Languages\Python\envs\py36\lib\site-packages\py\_io\terminalwriter.py", line 333, in write_out
    fil.write(msg)
  File "D:\Coding\Languages\Python\envs\py36\lib\site-packages\colorama\ansitowin32.py", line 40, in write
    self.__convertor.write(text)
  File "D:\Coding\Languages\Python\envs\py36\lib\site-packages\colorama\ansitowin32.py", line 141, in write
    self.write_and_convert(text)
  File "D:\Coding\Languages\Python\envs\py36\lib\site-packages\colorama\ansitowin32.py", line 169, in write_and_convert
    self.write_plain_text(text, cursor, len(text))
  File "D:\Coding\Languages\Python\envs\py36\lib\site-packages\colorama\ansitowin32.py", line 174, in write_plain_text
    self.wrapped.write(text[start:end])
OSError: [WinError 87] Paramètre incorrect
Exception ignored in: <_io.TextIOWrapper name='<stdout>' mode='w' encoding='utf-8'>
OSError: [WinError 87] Paramètre incorrect6
```

**Comment by Jan200101** [comment]
Works fine for me

**Comment by MatthieuDartiailh** [comment]
Actually it appears to be kind of random. Sometimes running my project testsuite works well and sometimes it fails in the middle because of this issue.

**Comment by eric-wieser** [comment]
Can you please edit that traceback and put it in a code block surrounded with ` ``` `?

**Comment by MatthieuDartiailh** [comment]
Done

**Comment by segevfiner** [comment]
This should be fixed for now by https://github.com/pytest-dev/pytest/pull/2462.

Note that colorama breaks independently of this for the same reason that this happens. See https://github.com/tartley/colorama/pull/131. It just stops working silently though since colorama doesn't check for errors on the WinAPI console functions it calls.

**Comment by SteveDoyle2** [comment]
Is there any progress on incorporating the bug fix (or an idea on how to work around it)?  I'm running into this issue intermittently on Windows with Python 2.7.13 and colorama 0.3.9.

**Comment by jsnowacki** [comment]
Likewise, I've observed it on both python 2.7 and 3.6 using VSCode terminal. I only noticed it while using `multiprocessing.Pool` and writing onto terminal. Thus, there may be some race condition for the IO.

**Comment by SteveDoyle2** [comment]
I am not using multiprocessing at all.  I've also seen a simple print inside colorama with no special characters fail, which is very weird.

I did add a try/except print the string I was having troble with.  It catches once out of ~100 prints.  It's a file path, so could the slashes be conflicting with each other?

**Comment by nicoddemus** [comment]
I believe this can be closed now that #131 has been merged.
